src/main/java/fr/insalyon/creatis/gasw/executor/batch/internals/BatchManager.java

```diff
@@ -0,0 +1,191 @@
+        final List<RemoteFile> filesDownload = new ArrayList<>();
+
+        jobData.setWorkingDir(wDirectoryJob);
+        jobData.setCommand(command);
+
+        filesUpload.add(new RemoteFile("./inv/" + jobID + "-invocation.json", wDirectoryJob));
+        filesUpload.add(new RemoteFile("./config/" + jobID + "-configuration.sh", wDirectoryJob));
+        filesUpload.add(new RemoteFile("./sh/" + jobID + ".sh", wDirectoryJob + "/sh"));
+        filesUpload.add(new RemoteFile("./workflow.json", wDirectoryJob));
+
+        filesDownload.add(new RemoteFile(jobData.getStderrPath(), "./err/" + jobID + ".sh.err"));
+        filesDownload.add(new RemoteFile(jobData.getStdoutPath(), "./out/" + jobID + ".sh.out"));
+        filesDownload.add(new RemoteFile(wDirectoryJob + jobID + ".sh.provenance.json", "./" + jobID + ".sh.provenance.json"));
+        filesDownload.add(new RemoteFile(wDirectoryJob + jobID + ".batch", "./" + jobID + ".batch"));
+
+        jobData.setFilesDownload(filesDownload);
+        jobData.setFilesUpload(filesUpload);
+
+        submitter(new BatchJob(jobData));
+    }
+
+    public BatchJob getJob(final String jobID) {
+        return jobs.stream()
+                .filter(job -> job.getData().getJobID().equals(jobID))
+                .findFirst()
+                .orElse(null);
+    }
+
+    public List<BatchJob> getUnfinishedJobs() {
+        return jobs.stream()
+                .filter(job -> !job.isTerminated())
+                .collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    @NoArgsConstructor
+    class BatchRunner implements Runnable {
+        private DateTime startedTime;
+
+        @Override
+        public void run() {
+            try {
+                startedTime = DateTime.now();
+                loop();
+            } catch (GaswException | InterruptedException e) {
+                log.error("Somehing bad happened during the K8sRunner", e);
+            }
+        }
+
+        private void sleep() throws GaswException, InterruptedException {
+            final Duration diff = new Duration(startedTime, DateTime.now());
+
+            if (diff.getStandardSeconds() > config.getOptions().getTimeToBeReady()) {
+                throw new GaswException("Volume wasn't eady in 2 minutes, aborting !");
+            } else {
+                Thread.sleep(GaswConfiguration.getInstance().getDefaultSleeptime());
+            }
+        }
+
+        private void loop() throws InterruptedException, GaswException {
+            while (initialized != true) {
+                sleep();
+            }
+            while (end == false) {
+                synchronized (this) {
+                    for (final BatchJob exec : getUnfinishedJobs()) {
+                        if (exec.getStatus() == GaswStatus.NOT_SUBMITTED) {
+                            exec.start();
+                        }
+                    }
+                }
+                Thread.sleep(GaswConfiguration.getInstance().getDefaultSleeptime());
```

Comment:
I do not like the error management here.
If one job submission fails, the whole thread ends and nothing happens anymore (gasw / moteurLite / etc will be stucked.

First we can do as in dirac, catch and log GaswException on submit, retrying later.
But it will try indefinitely, not pretty.
We can also not be tolerant at all, catching the exception and telling moteurLite it has failed.